session6/src/player.py

```diff
@@ -9,30 +9,52 @@
 class Player(pygame.sprite.Sprite):
     def __init__(self):
         super().__init__()
-        self.x_speed = 5  # speed in the x direction
-        self.y_speed = 0  # speed in the y direction
-        image_location = os.path.join(
-            "assets", "player.png"
-        )  # find the path of the player image
+
+        # Load the player image, set class rectangle to that of the image
+        image_location = os.path.join("assets", "player.png")  # Find the path of the player image
         self.image = pygame.image.load(image_location).convert_alpha()  # load the image
         self.rect = self.image.get_rect()  # get the rect of the image
+        
+        # Set the initial position of the player
         self.rect.x = 400  # have the player start at 400 on x axis
+        self.rect.y = 0 # this is initialized for us (self.image.get_rect()), but you can change it if you want
 
+        # Physics       
         self.jump_cooldown = 0  # the player can jump right away!
+        # speed (change in position per tick)
+        self.x_speed = 5  # speed in the x direction (must be greater than 0, or you won't move!)
+        self.y_speed = 0  # speed in the y direction 
 
     # how do we want to implement the other actions?
 
-    def update(self, keys_pressed):
+    # Parameters:
+    #   - keyspressed - result of pygame.key.get_pressed(), passed in from main
+    def update(self, keys_pressed, mouse_buttons, mouse_pos):
+
+        # If you want, feel free to seperate the sections out into different methods.
+        # I prefer to keep it all together so you don't have to worry about how it fits together.
+
+        # SETUP  -------------------------------------
+
+        teleport = False # must reset to False, or else the player will keep teleporting to the cursor after the first click
+        direction = [0, 0] # direction for this frame
 
-        # Setup
+        # End of SETUP
 
-        print("y speed: " + str(self.y_speed))
-        # Events -> move these in from the main loop
+        # EVENTS -------------------------------------
 
-        # Get input
+        # Recall: keys_pressed is passed to us from the main loop
+        # You can check for keys pressed as usual!
+
+
+        # Check if the player needs to teleport
+        if mouse_buttons[0]:  # If left mouse pressed
+            teleport = True  # Replace this line
```

Comment:
Get rid of outdated comment
```suggestion
            teleport = True
```

Reply:
Resolved in PR #28